067-kraft-jbod-support.md

```diff
@@ -0,0 +1,221 @@
+It is also possible to not set this option for any volume as well.
+In that case, the operator will decide which volume should be used for the metadata log on its own.
+
+Following example shows how the configuration might look like in order to use the second volume to store the metadata log:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: broker
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - broker
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+      - id: 1
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+        kraftMetadata: shared
+```
+
+The reason why the `kraftMetadata` property is not just a boolean is to make it possible to extend the support in the future.
+One such possibility would be to support dedicated metadata volumes, which is discussed in a separate section.
+
+The `kraftMetadata` option will be ignored when used with ZooKeeper-based clusters.
+
+### Kafka configuration
+
+Currently, we only use the `log.dirs` option to configure the log directories.
+With this proposal, the `log.dirs` option will be configured in the same way as present.
+In addition to that, we will also configure the `metadata.log.dir` option that we do not use today.
+This option will be configured only for KRaft clusters.
+It will be set to the volume marked as `kraftMetadata: shared`.
+Or - if the flag is not set for any of the volumes - it will be set to the volume with the lowest configured ID.
+
+The currently used `metadata.log.dir` will be also stored in an annotation on the `StrimziPodSet` representing the node pool.
+This annotation will be used to detect changes to the metadata log volume and used to log such events.
+In the future, if necessary, it could also be used, for example, to prevent changes or to reassign the metadata log in a different way.
+However, such functionality is not planned as part of this proposal.
+
+### Changing the metadata log directory / volume
+
+The volume / directory used for the metadata log might change in several situations:
+* When a user changes the `kraftMetadata` flag and moves it to another volume
+* When the volume originally used for the metadata is removed
+* When a new volume with lower volume ID is added to the JBOD list and the `kraftMetadata` flag is not set
+
+Changes to the `metadata.log.dir` always require rolling update of the affected Kafka nodes.
+The change of the directory will be handled in the Kafka start-up scripts in our container image.
+The scripts will do the following:
+1) Try to find the existing metadata log
+2) Compare the current location of the metadata log with the desired location
+3) If the locations differ, the old metadata log will be deleted
```

Comment:
Question: How would we delete the old metadata log while the same log dir contains other data logs? Only deleting the `__cluster_metadata` dir and `meta.properties` file?

Reply:
I don't remember exactly from when I was testing it - I think deleting the `__cluster_metadata-0` directory was enough.

Reply:
in any case, it would delete just the directories / files related to metadata and not the data related to regular message logs.

Reply:
Yes, that's what I thought, and that should be fine as long as we don't touch other data log files.

Reply:
I updated the bullet point to make it more clear.